### PR TITLE
Flip autoconnected slots to explicit connections, pt 1

### PR DESCRIPTION
**Pull request by nyalldawson**
Using https://phabricator.kde.org/D8187, this PR flips a bunch of autoconnected slots to explicit connections.

Why?

Because Qt autoconnected slots were a bad idea. They are extremely fragile and give no build or run time errors when the connection is broken. They make refactoring dialogs a pain. They don't work with IDEs when you search for uses of a widget.

Basically, s*** all round. Let's get rid of them and have a better future.

In fact, they're such a bad idea, I suspect Qt will drop them sometime. Better to dump them before they do.

### CODE REVIEW

File: python/gui/qgsrasterformatsaveoptionswidget.sip [hunk 1]
```diff
@@ -104,17 +104,17 @@ class QgsRasterFormatSaveOptionsWidget: QWidget
 %End
 
   private slots:
-    void on_mProfileNewButton_clicked() ;
+    void mProfileNewButton_clicked() ;
```
Comment (m-kuhn) [inline python/gui/qgsrasterformatsaveoptionswidget.sip]: Wow, that's a lot of `private slots` here :D

**Review by m-kuhn — APPROVED** [review]
Yes, please

File: python/gui/qgsrasterformatsaveoptionswidget.sip (hunk 1 above)
Comment (nyalldawson) [reply python/gui/qgsrasterformatsaveoptionswidget.sip]: Hmm... yes.... @3nids any idea why these are sneaking into the bindings?

Comment (3nids) [reply python/gui/qgsrasterformatsaveoptionswidget.sip]: @nyalldawson should be fixed by #5328

**Pull request merged**